src/du_parameters/afrcn.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""ARFCN calculations for different frequencies."""
+
+from dataclasses import dataclass
+from typing import Optional, Union
+
+
+@dataclass
+class ARFCNRange:
+    """ARFCN range class."""
+
+    lower: float
+    upper: float
+    freq_grid: float
+    freq_offset: float
+    arfcn_offset: int
+
+
+LOW = ARFCNRange(0, 3000, 0.005, 0, 0)
+MID = ARFCNRange(3000, 24250, 0.015, 3000, 600000)
+HIGH = ARFCNRange(24250, 100000, 0.06, 24250, 2016667)
+
+
+def freq_to_arfcn(frequency: Union[int, float]) -> Optional[int]:
```

Comment:
```suggestion
def freq_to_arfcn(frequency: Union[int, float]) -> int:
```

This is only returning an integer. When there is an error, it is raising an exception, not returning.

Reply:
removed.